.github/workflows/ci-additional.yaml

```diff
@@ -116,68 +133,23 @@ jobs:
           python xarray/util/print_versions.py
       - name: Install mypy
         run: |
-          python -m pip install "mypy==1.13" --force-reinstall
+          python -m pip install mypy pytest-mypy-plugins
 
       - name: Run mypy
         run: |
           python -m mypy --install-types --non-interactive --cobertura-xml-report mypy_report
 
-      - name: Upload mypy coverage to Codecov
-        uses: codecov/codecov-action@v5.3.1
-        with:
-          file: mypy_report/cobertura.xml
-          flags: mypy
-          env_vars: PYTHON_VERSION
-          name: codecov-umbrella
-          fail_ci_if_error: false
-
-  mypy-min:
-    name: Mypy 3.10
-    runs-on: "ubuntu-latest"
-    needs: detect-ci-trigger
-    defaults:
-      run:
-        shell: bash -l {0}
-    env:
-      CONDA_ENV_FILE: ci/requirements/environment.yml
-      PYTHON_VERSION: "3.10"
-
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0 # Fetch all history for all branches and tags.
-
-      - name: set environment variables
+      - name: Run mypy tests
+        # Run pytest with mypy plugin even if mypy analysis in previous step fails.
+        if: ${{ always() }}
         run: |
-          echo "TODAY=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
-      - name: Setup micromamba
-        uses: mamba-org/setup-micromamba@v2
-        with:
-          environment-file: ${{env.CONDA_ENV_FILE}}
-          environment-name: xarray-tests
-          create-args: >-
-            python=${{env.PYTHON_VERSION}}
-          cache-environment: true
-          cache-environment-key: "${{runner.os}}-${{runner.arch}}-py${{env.PYTHON_VERSION}}-${{env.TODAY}}-${{hashFiles(env.CONDA_ENV_FILE)}}"
-      - name: Install xarray
-        run: |
-          python -m pip install --no-deps -e .
-      - name: Version info
-        run: |
-          python xarray/util/print_versions.py
-      - name: Install mypy
-        run: |
-          python -m pip install "mypy==1.13" --force-reinstall
-
-      - name: Run mypy
-        run: |
-          python -m mypy --install-types --non-interactive --cobertura-xml-report mypy_report
+          python -m pytest -v --mypy-only-local-stub --mypy-pyproject-toml-file=pyproject.toml xarray/**/test_*.yml
```

Comment:
Can we move these args into a config file, so it's extremely easy to run locally? Ideally when just running `pytest`...

Reply:
I originally planned to add these to the pytest addopts setting in pyproject.toml, but wasn't sure if that would end up interfering with something I'm not yet aware of. Is that where you'd like me to move these?

Reply:
Yes please!